test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -209,6 +209,20 @@ func (suite *k8sSuite) TestAdmissionControllerWebhooksExist() {
 		}
 		suite.Require().True(found, fmt.Sprintf("None of the mutating webhook configurations have the name '%s'", expectedWebhookName))
 	})
+
+	suite.Run("agent registered validating webhook configuration", func() {
+		validatingConfigs, err := suite.K8sClient.AdmissionregistrationV1().ValidatingWebhookConfigurations().List(ctx, metav1.ListOptions{})
+		suite.Require().NoError(err)
+		suite.NotEmpty(validatingConfigs.Items, "No validating webhook configuration found")
+		found := false
+		for _, validatingConfig := range validatingConfigs.Items {
+			if validatingConfig.Name == expectedWebhookName {
+				found = true
+				break
+			}
+		}
+		suite.Require().True(found, fmt.Sprintf("None of the validating webhook configurations have the name '%s'", expectedWebhookName))
+	})
```

Comment:
Why `List`ing all webhooks and iterating over the result to find the expected one?
Could we just try to `Get` the expected one?

```suggestion
	suite.Run("agent registered validating webhook configuration", func() {
		validatingConfig, err := suite.K8sClient.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, expectedWebhookName, metav1.GetOptions{})
		suite.Require().NoError(err)
		suite.NotNilf(validatingConfig, "None of the validating webhook configurations have the name '%s'", expectedWebhookName)
	})
```

Reply:
Good catch. I didn't realize that `Get` was in the API. I've also updated the mutating webhook test just above to also use this more concise way. Thanks for calling this out!